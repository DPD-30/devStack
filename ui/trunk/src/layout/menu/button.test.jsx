import React from 'react'
import { renderToString } from 'react-dom/server'
import { HashRouter as Router } from 'react-router-dom'

import LayoutMenuButton from './button.jsx'

jest.unmock('./button.jsx')


describe('', () => {
    it('LayoutMenuButton should match snapshot', () => {
        const comp = renderToString(
            <Router>
                <LayoutMenuButton
                    label="MockLabel"
                    url="/mocked"
                    name="mockedMenuButton"
                />
            </Router>
        )

        expect(comp).toMatchSnapshot()
    })
    it('LayoutMenuButton should match snapshot for active item', () => {
        const comp = renderToString(
            <Router>
                <LayoutMenuButton
                    label="Home"
                    url="/home"
                    name="home"
                />
            </Router>
        )

        expect(comp).toMatchSnapshot()
    })
})
