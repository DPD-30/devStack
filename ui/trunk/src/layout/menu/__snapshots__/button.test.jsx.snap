// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutMenuButton should match snapshot 1`] = `
<li class="nav-item">
  <a name="mockedMenuButton"
     class="nav-link"
     href="#/mocked"
  >
    MockLabel
  </a>
</li>
`;

exports[`LayoutMenuButton should match snapshot for active item 1`] = `
<li class="nav-item">
  <a name="home"
     class="nav-link"
     href="#/home"
  >
    Home
  </a>
</li>
`;
